routers/web/web.go

```diff
@@ -1286,6 +1286,7 @@ func registerRoutes(m *web.Route) {
 			m.Post("/cleanup", context.RepoMustNotBeArchived(), context.RepoRef(), repo.CleanUpPullRequest)
 			m.Group("/files", func() {
 				m.Get("", context.RepoRef(), repo.SetEditorconfigIfExists, repo.SetDiffViewStyle, repo.SetWhitespaceBehavior, repo.SetShowOutdatedComments, repo.ViewPullFiles)
+				m.Get("/{sha}", context.RepoRef(), repo.SetEditorconfigIfExists, repo.SetDiffViewStyle, repo.SetWhitespaceBehavior, repo.SetShowOutdatedComments, repo.ViewPullFiles)
```

Comment:
How about to use a query parameter, then we don't need to copy the line.

Reply:
I tried to use the same Uri style as GitHub (so in future also a range of commits could be added)

Reply:
i just did some more testing with the github urls and will try to align the urls as much as possible

github uses 3 different ways:
- show single commits diff in PR context: /pull/{pull-id}/commits/{sha1}
- show diff of all commits since (of PR) :  /pull/{pull-id}/files/{sha1}
- show diff of a commit range (of PR): /pull/{pull-id}/files/{sha1}..{sha1}

Reply:
Definitely prefer GH-style URLs.

Reply:
Yes - i just added the routes.

The commit selection dropdown gets more complex now, so i think i'll transfer it to a vue component.

Reply:
I guess that's fine. Generally I'm wary of new Vue stuff because they cause loading artifacts, but given that the dropdown is not rendered on page load, this one is safe to convert.

Reply:
I'd prefer a reverse logic of the second route (show all changes in PR from start till specified commit id). 
I think the use case is more often "show all changes since this commit" - for example after you did a review and only want to see new files / changes.

Reply:
But it's ok for me - i'll change the logic for this case.